### DISCUSSION TITLE
Licensing

### DISCUSSION THREAD

**Discussion by arelath** [gildor2/UEViewer#254] [General]
I'm currently working as a software engineer at Electronic Arts using the unreal engine. Unreal is pretty new to me, so I'm still learning. We'd like to edit assets at runtime, but the unreal license prohibits shipping their editing tools in any way, so we need to come up with solutions that don't use their editing code. 

Using UEViewer as a starting point is an interesting idea (haven't even prototyped anything yet), but not having any license means there's no way we could use any of your code. No license is the most restrictive license you can have.

Since you said BSD-3 clause looked good but you didn't like the header data, you might look into using the MIT license (https://en.wikipedia.org/wiki/MIT_License). It's pretty similar to BSD-3, but doesn't require header data on your part. 

Sorry, I can't really disclose what I'm doing due to my NDA being very specific. This is just one possible path we could take. Also I need to learn more about what I can and can't do with Unreal.

**Comment by gildor2** [comment]
Yeah, I was too lazy adding any license. If you're familiar with this stuff, and can briefly describe which steps should be done for it, I'll add the license.

BTW it's very nice to hear that EA is interested in my project :) This makes me proud of it!

**Reply by arelath** [reply]
You should be proud. A lot of work went into this project and as far as I can tell, it's good quality code (I haven't done a deep dive on the code or run it long enough to find bugs). I'm all for the moding community and would love it if Epic Games opened up their engine more to the moders. It's just becomes a problem for online cheating and piracy.    

Adding a license is really simple on GitHub and takes less than 5 minutes. Here's the instructions: https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository

**Comment by gildor2** [comment]
Added the MIT license
